qos/jsonrpc/request.go

```diff
@@ -23,30 +24,27 @@ type Request struct {
 	Params  Params  `json:"params,omitempty"`
 }
 
-// NewRequest constructs a new JSONRPC request from the given ID, method, and optional parameters.
+// BuildParamsFromStrings builds a Params object from an array of strings.
 //
-// For example:
-// id - 1
-// method - eth_getBalance
-// params - ["0x28C6c06298d514Db089934071355E5743bf21d60", "0xe71e1d"]
+// For example, for an `eth_getBalance` request, the params would look like:
+// params - ["0x28C6c06298d514Db089934071355E5743bf21d60", "0xe71e1d"]]
 //
-// The above request would be serialized as:
-// {"jsonrpc":"2.0","id":1,"method":"eth_getBalance","params":["0x28C6c06298d514Db089934071355E5743bf21d60", "0xe71e1d"]}
-func NewRequest(id int, method Method, params ...any) Request {
-	request := Request{
-		JSONRPC: Version2,
-		ID:      IDFromInt(id),
-		Method:  method,
-	}
-
-	if len(params) > 0 {
-		jsonParams, err := json.Marshal(params)
-		if err == nil {
-			request.Params = Params{rawMessage: jsonParams}
+// JSON-RPC array params must be passed in the order specified by the method.
+// Reference: https://www.jsonrpc.org/specification#parameter_structures
+//
+// TODO_FUTURE(@commoddity): other helper methods may be required to build
+// params for different JSON-RPC methods, eg. ["<string>", <bool>]
+func BuildArrayParamsFromStrings(params [2]string) (Params, error) {
```

Comment:
Not a blocker: 
Either:
-  Rename the method to emphasize it takes an array (and not a slice)
- Change the `params` type to `[]string`

Reply:
Done.